config/provider.go

```diff
@@ -12,8 +27,7 @@ const (
 )
 
 // GetProvider returns provider configuration
-func GetProvider() *tjconfig.Provider {
-	resourceMap := tf.Provider().ResourcesMap
+func GetProvider(tf *schema.Provider) *tjconfig.Provider {
```

Comment:
Could you add some description to the PR especially regarding the motivation behind this line.

Reply:
Not a big motivation honestly, I just thought accepting it as argument from `main.go` would allow us to write tests for this function with synthetic TF schemas and also not have to compile the actual provider during those test runs. Maybe make easier to transition to https://github.com/crossplane-contrib/terrajet/issues/131